### ISSUE TITLE
screenFetch recognizes XMonad as i3

### ISSUE THREAD

**Issue by FrostyX** [KittyKatt/screenFetch#133]
Hi, just tried screenFetch and it recognizes XMonad as i3.

	    .:+sydNMMMMMNNNmmmdddhhhhhhmMmy   DE: Not Present
	       /mMMMMMMNNNmmmdddhhhhhmMNhs:   WM: i3

but

	$[FrostyX  ~]-> ps ax |grep i3
	16008 pts/6    S+     0:00 grep --colour=auto i3
	26350 ?        S      4:17 /home/frostyx/.xmonad/xmonad-i386-linux ...

**Comment by haasn** [comment]
This change breaks detection on xmonad, ironically.

The process name, at least on my machine, is ```xmonad-x86_64-linux```, but since pgrep now searches for ```^xmonad$``` it doesn't find anything.

Perhaps it would be better to search for ```^$each``` instead of ```^$each$```? Alternatively, it might be best to change the regex for ```xmonad``` to ```xmonad-.*```.

**Comment by neuschaefer** [comment]
Sorry for breaking xmonad detection. @haasn, I'd go with changing the match string to `xmonad-*`. A few notes though:
* `.*` will not work because these aren't "proper" regular expressions, but glob matches, which are evaluated by bash. `*` does seem to work.
* Wildcards in `case` matches do not seem to work with quotes, so instead of `'xmonad-*')`, it would have to be `xmonad-*)`.

**Comment by KittyKatt** [comment]
Argh, getting busy busy busy. Time might be coming my way soon, though. I'll definitely take a look at all of these issues.

**Comment by GLolol** [comment]
Does GLolol@b491d48 change anything?

**Comment by FrostyX** [comment]
@GLolol still not working

    -> git clone https://github.com/GLolol/screenFetch.git
    -> cd screenFetch
    -> git checkout b491d48607f3beda21cf60c822fa2f017b9200dc
    -> ./screenfetch-dev |grep WM
      :oyhhdNNMMMMMMMNNNmmdddhhhhhyymMh   WM: LG3D

**Comment by GLolol** [comment]
That is weird... I don't see LG3D anywhere in the code.

**Comment by GLolol** [comment]
Okay, I tried something else; I didn't notice WM detection was in two places instead of one (I'm editing with a phone on a tiny screen...)

**Comment by FrostyX** [comment]
I am sorry, I won't make you happy

    -> git pull
    -> git checkout bdaba13878c3987bf8beb91aa8fc0ae745bbedc1
    -> ./screenfetch-dev |grep WM
      :oyhhdNNMMMMMMMNNNmmdddhhhhhyymMh   WM: LG3D

**Comment by GLolol** [comment]
@FrostyX I did some digging on the interwebs... [Have you been using Java and SetWMName?](https://wiki.haskell.org/Xmonad/Frequently_asked_questions#Using_SetWMName)

**Comment by GLolol** [comment]
Also, XMonad detection works fine on my setup, both with and without my testing patches! (Debian Jessie with xmonad from official repos)

![](https://dl.dropboxusercontent.com/u/18664770/screenfetch%2Bxmonad.PNG)

**Comment by GLolol** [comment]
Information affecting this and #185:

1) XMonad compiles manual configurations in `~/.xmonad`, with a binary name that reflects the current OS and architecture: e.g. `xmonad-i386-linux` or `xmonad-x86_64-linux`. For some reason, screenFetch fails to detect this, but a plain `/usr/bin/xmonad` install (like the one above) works fine.

2) XMonad has the ability to disguise itself as other window managers using [SetWMName](https://wiki.haskell.org/Xmonad/Frequently_asked_questions#Using_SetWMName), which seems to change what it shows in `_NET_WM_NAME`

3) Because screenfetch can't find xmonad from `xmonad-i386-linux` or `xmonad-x86_64-linux`, it starts doing some weird things. Since `i3` matches 'xmonad-**i3**86-linux', it will display the WM as i3 for those on x86. Other times, it will start falling back to `_NET_WM_NAME` in the case of x64, which may not always be accurate because of the SetWMName hack.

In short, this doesn't have much to do with i3 detection, but the XMonad globs that are broken!
